staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -632,7 +669,15 @@ func (s *store) List(ctx context.Context, key, resourceVersion string, pred stor
 				return storage.NewInternalErrorf("unable to transform key %q: %v", kv.Key, err)
 			}
 
-			if err := appendListItem(v, data, uint64(kv.ModRevision), pred, s.codec, s.versioner, newItemFunc); err != nil {
+			cluster := genericapirequest.ClusterFrom(ctx)
+			clusterName := ""
+			if cluster != nil && cluster.Wildcard {
```

Comment:
Add a comment that you are extracting the cluster from the key (because we don't want to store the cluster name in etcd) and that the key is the only thing that links us to a cluster.